org.eclipse.wildwebdeveloper.xml/src/org/eclipse/wildwebdeveloper/xml/XMLLSClassPathExtensionRegistry.java

```diff
@@ -0,0 +1,61 @@
+package org.eclipse.wildwebdeveloper.xml;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.runtime.IConfigurationElement;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.Status;
+
+public class XMLLSClassPathExtensionRegistry {
```

Comment:
Actually, I'm not sure you really need to keep a copy of the model here. Maybe it's easier and more powerful and cheap enough to query the extension registry everytime it's needed. WDYT?

Reply:
I'm not entirely sure I understand. Do you mean we should do without the "sync extension registry" pattern here? That should be ok, as in this implementation we only need to query the extension registry when the XML LS is launched.

I looked into adding onto the classpath dynamically (after the LS would be launched) and it seems to be possible but [involves using class loaders and reflection.](https://community.oracle.com/thread/1239849) Im not sure if there are any use cases that would merit dynamically adding onto the classpath. Thus it should be safe to say that the xtension registry is only needed when launching the XML LS.

Reply:
> Do you mean we should do without the "sync extension registry" pattern here?

Yes, but actually, we can just keep it as it at the moment; unless you think it's worth changing it right now.

> I looked into adding onto the classpath dynamically (after the LS would be launched) 

Adding to classpath after load is too hard, and as you identified, there isn't use-case for it.
I don't think we need more dynamic handling. My suggestion was more about whether we do really need an sync'd cache for this case or whether just querying the extension registry on every start of the LS would be simpler.

Reply:
Ok sounds good. Lets keep it for the moment (and fix it before the merge) but I agree there's no reason to cache the extension registry. This is probably also true for the xmllsExtension/XMLExtension registry (but that's another topic).